components/fxa-client/src/internal/state_manager.rs

```diff
@@ -28,6 +31,9 @@ pub struct StateManager {
     auth_state: AuthState,
     /// Application-supplied storage handlers
     storage_handler: Option<Box<dyn StorageHandler>>,
+    /// Application-supplied event listener.  Wrapped with an Arc since it's shared with
+    /// FirefoxAccount
+    pub event_listener: Option<Arc<Box<dyn EventListener>>>,
```

Comment:
This feels a little like a leaking abstraction, what do you think about keeping the event notification at a higher level than the state manager, and keeping the state manager only responsible for writing and reading state?

Reply:
I think this goes well with your comment above...
  - `StateManager` could have methods to check if the state has changed since some time in the past -- one to check if the any persisted state has changed and one to check if the auth state has changed.
  - The top-level `FirefoxAccount` would wrap each API call with state-change checks
  - If the persisted state changed, then it would call `StorageHandler::save_state()`
  - If the auth state changed, then it would pass an `AuthStateChanged` event `EventHandler::on_event()`